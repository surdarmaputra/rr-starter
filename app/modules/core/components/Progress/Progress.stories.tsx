import type { Meta, StoryObj } from '@storybook/react';

import { Progress, type ProgressProps } from './Progress';

type Story = StoryObj<typeof Progress>;

const meta: Meta = {
  component: Progress,
  title: 'Core/Components/Progress',
};

export default meta;

export const Default: Story = {
  args: {
    value: 45,
  },
  render: (args: ProgressProps) => <Progress {...args} />,
};
