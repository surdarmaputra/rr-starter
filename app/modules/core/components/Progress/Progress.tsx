import * as ProgressPrimitive from '@radix-ui/react-progress';
import * as React from 'react';

import { mergeClassNames } from '~/modules/core/libs/utils/ui';

export type ProgressProps = React.ComponentPropsWithoutRef<
  typeof ProgressPrimitive.Root
>;

const Progress = React.forwardRef<
  React.ComponentRef<typeof ProgressPrimitive.Root>,
  ProgressProps
>(({ className, value, ...props }, ref) => (
  <ProgressPrimitive.Root
    className={mergeClassNames(
      'relative h-4 w-full overflow-hidden rounded-full bg-primary-disabled',
      className,
    )}
    ref={ref}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className="h-full w-full flex-1 animate-pulse rounded-full bg-primary transition-all duration-1000"
      style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
    />
  </ProgressPrimitive.Root>
));
Progress.displayName = ProgressPrimitive.Root.displayName;

export { Progress };
