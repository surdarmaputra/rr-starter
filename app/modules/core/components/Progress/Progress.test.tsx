import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Progress } from './Progress';

describe('Progress', () => {
  function renderComponent() {
    return render(<Progress data-testid="Progress-testid" value={50} />);
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Progress-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
