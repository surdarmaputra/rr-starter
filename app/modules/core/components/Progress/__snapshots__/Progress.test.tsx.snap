// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress > renders snapshot 1`] = `
<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    class="relative h-4 w-full overflow-hidden rounded-full bg-primary-disabled"
    data-max="100"
    data-state="indeterminate"
    data-testid="Progress-testid"
    role="progressbar"
  >
    <div
      class="h-full w-full flex-1 animate-pulse rounded-full bg-primary transition-all duration-1000"
      data-max="100"
      data-state="indeterminate"
      style="transform: translateX(-50%);"
    />
  </div>
</div>
`;
